src/llm/http_llm_calculator.cc

```diff
@@ -664,22 +725,26 @@ std::string HttpLLMCalculator::serializeStreamingChunk(const std::string& chunkR
     // logprobs: object/null; Log probability information for the choice. TODO
     writer.String("logprobs");
     writer.Null();
-    // delta: object
-    writer.String("delta");
-    writer.StartObject();  // {
-    // content: string; Actual content of the text produced
-    if (!stop) {
-        writer.String("content");
-        writer.String(chunkResponse.c_str());
+    if (endpoint == CHAT_COMPLETIONS) {
+        writer.String("delta");
+        writer.StartObject();  // {
+        if (!stop) {
+            writer.String("content");
+            // writer.String("role");
+            // writer.String("assistant");
+            // role: string; Role of the text producer
+            // Will make sense once we have chat templates? TODO(atobisze)
+            writer.String(chunkResponse.c_str());
+        }
+        writer.EndObject();  // }
+    } else if (endpoint == COMPLETIONS) {
+        if (!stop) {
```

Comment:
Don't we need to start with "delta" here as well? 
@dkalinowski

Reply:
No, this legacy api has no deltas:
https://platform.openai.com/docs/api-reference/completions/create